src/Security/Authentication/Core/src/RemoteAuthenticationHandler.cs

```diff
@@ -186,16 +186,25 @@ protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
         protected override Task HandleForbiddenAsync(AuthenticationProperties properties)
             => Context.ForbidAsync(SignInScheme);
 
+        /// <summary>
+        /// Creates a 32 bit random Id and Base64Url encodes it.
+        /// </summary>
+        /// <returns></returns>
+        protected virtual string GenerateUniqueId()
```

Comment:
Who overrides this? Is there a test that overrides this?

Reply:
Nobody at the moment, but this base class is intended to allow as much extensibility as is practical.

Reply:
Its probably worth considering just moving it all the way up to AuthenticationHandler then, as its just a general purpose helper at that point

Reply:
There are specific requirements around it for PKCE, so not sure a general unique id base API would always work. FWIW.

Reply:
If that's the case maybe add a specific GeneratePKCEId method in OAuthHandler

Reply:
I had generalized this one because it was already fulfilling the PKCE requirements. I can revert that part and duplicate the code to be more explicit.